closed/src/java.base/share/native/libjncrypto/NativeCrypto.c

```diff
@@ -440,6 +455,27 @@ JNIEXPORT jlong JNICALL Java_jdk_crypto_jniprovider_NativeCrypto_loadCrypto
         }
     }
 
+    /* Check whether the loaded OpenSSL library is in FIPS mode. */
+    if (ossl_ver >= OPENSSL_VERSION_3_0_0) {
+        typedef int OSSL_fipsmode_t(OSSL_LIB_CTX *);
+        OSSL_fipsmode_t* OSSL_fipsmode;
+        OSSL_fipsmode = (OSSL_fipsmode_t*)find_crypto_symbol(crypto_library, "EVP_default_properties_is_fips_enabled");
+        if ((NULL != OSSL_fipsmode) && ((*OSSL_fipsmode)(NULL) == 1)) {
```

Comment:
Suggestions:
* this local should be initialized in its declaration
* names beginning with a capital suggest they're global (this is not)
* missing space before `*` in pointer types
* constant belong on the left of `==` and `!=`
```c
        typedef int OSSL_fipsmode_t(OSSL_LIB_CTX *);
        OSSL_fipsmode_t *ossl_fipsmode = (OSSL_fipsmode_t *)find_crypto_symbol(crypto_library, "EVP_default_properties_is_fips_enabled");
        if ((NULL != ossl_fipsmode) && (1 == (*ossl_fipsmode)(NULL))) {
```
Similar comments for lines 470-472.

Reply:
Updated in both spots.

Reply:
> names beginning with a capital suggest they're global (this is not)

That part has not been addressed (i.e. `ossl_fipsmode` is a better name than `OSSL_fipsmode`).

Reply:
Sorry, I missed that. Changed to suggestion.